source/Hardpoint.cpp

```diff
@@ -39,9 +39,30 @@ namespace {
 
 
 // Constructor.
-Hardpoint::Hardpoint(const Point &point, const Angle &baseAngle, bool isTurret, bool isParallel, const Outfit *outfit)
-	: outfit(outfit), point(point * .5), baseAngle(baseAngle), isTurret(isTurret), isParallel(isParallel)
+Hardpoint::Hardpoint(const Point &point, const vector<Angle> &angles, bool isTurret, bool isParallel, const Outfit *outfit)
+	: outfit(outfit), point(point * .5), isTurret(isTurret), isParallel(isParallel)
 {
+	if(isTurret)
```

Comment:
I prefer a solution where we don't need to add a body to the constructor (and I think that is possible).